### DISCUSSION TITLE
V3 Documentation STILL under construction

### DISCUSSION THREAD

**Discussion by ouijahija** [shinyorg/shiny#1526] [Q&A] [answered]
First of I wanna thank shinyorg for the tremendous work on this library. I'm using it quite a lot and it works quite well!

Now, my main gripe is, that the documentation on shinylib.org (specifically BluetoothLE and Local Notifications, as these are the ones I use the most) is still under construction. Of course there are the code samples, where you can look up some of the use cases. But having to sift through all the scripts that might have the answer that I'm looking for is cumbersome and there might not even be a sample for what I'm looking for! Plus it's extremely frustrating if you're wondering how to use the library and the the documentation only says 'TODO', so you have to find it out by trial-and-error or hoping there's a sample for it. The documentation for BluetoothLE for example is since over a half year still quite bare bones and in some places even contains incorrect code...

I tried to find any updates on the documentation but didn't find any, so I'm asking it here: **When can we expect any updates to the documentation? V4 might be released this year. Will the changes be that big that it makes the V3 documentation osbsolete?** Because if not, I don't see, why this takes so long to fill that in...

I'm kinda surprised, no one has asked this before. And admittedly it makes me kinda paranoid I missed some very obvious place where I can look up the answer. Sorry if that's the case!

**Comment by aritchie** [comment, marked as answer]
> Now, my main gripe 

This is free code - you have no gripes.  You can contribute or go elsewhere

> When can we expect any updates to the documentation? 

You can't expect anything.  This is code I work on for myself first, everyone else second.  You consume it for free.

**Reply by ouijahija** [reply]
You're right... Rereading my original post I relize that sound quite entitled. I'm sorry for that!
Please know that I very much so appreciate the work you're doing and that I regret the tone I used.

**Comment by munkii** [comment]
Note I have never met @aritchie and with only a couple of short exchanges online I certainly don't know him.

@ouijahija you are right the docs are "still under construction" and sometimes I do think "FFS". I also struggle getting some of the samples to work but as Allan says it's free code. When I get over my issue do I think "Ooo someone else might have that issue I'll update the Shiny docs." No I do not I just move on.

Now that I have made it clear that I am no saint I'll continue and @ouijahija sorry I know this is not just you.

I'm assuming you are using Shiny because either you don't have the skillset to write it yourself or the time. It's both for me.

Have a guess how much time (which at the end of the day equals money) you (just like me and many, many others have saved using Shiny). Now answer this question "Do I sponsor the Shiny repo or any of the repos I use?" If the answer is No then ask yourself the following two questions

1. How much am I going to sponsor my favorite repos?
2. [There are no other questions....]

For the record, Allan won't be retiring on the sponsorship he gets from me, the amount is embarassingly small but if we all "sponsored emabrassingly small" amounts we'd have a much happier bunch of oss authors. and I firmly belive a happier @aritchie would be better for all of us ;-)
